packages/vuetify/src/components/VDialog/VDialog.sass

```diff
@@ -65,6 +65,8 @@
   overflow-y: auto
   top: 0
   left: 0
+  width: 100% !important
```

Comment:
We'll have to come up with a different solution that doesn't required **!important**

Reply:
What issue is caused by `!important` in this spot?

I'm not sure there is a different solution in just CSS, since the width is being applied as a direct style on the element, AFAIK only `!important` can override it.

Reply:
What I mean is, why do you need !important here? What does it solve?

Reply:
It's stated in the description for the PR:

> Use the .v-dialog--fullscreen class to enforce full width, rather than relying on removing the styles applied by the max-width and width props.

The element will have a `style` prop which has `width` and `max-width`. Due to the gotcha in Vue SSR described in the issue, the hydrated client is unable to remove those styles, leading to an odd state where the element has the `v-dialog--fullscreen` class but also the `style` prop. So the change with `!important` is to ensure that if the `v-dialog--fullscreen` class is applied, the dialog is indeed fullscreen.

Reply:
Then why not add the SSR check here: https://github.com/vuetifyjs/vuetify/blob/2b975f575f47ba75aa5d0dd9fba859595c7b9ba4/packages/vuetify/src/components/VDialog/VDialog.ts#L271

Reply:
I'm not sure what check you're proposing. If it would be a check to not apply the styles if on the server, then I believe that would fix the issue of getting into a broken state. However, it would cause the dialog to be rendered incorrectly until hydration completed. If the developer is aiming to have two possibilities for the dialog, either fullscreen or with width N, the server-rendered markup would be neither, as it would not apply the width style until hydration is done.

I wrote the PR as it is because it's a CSS solution, which keeps the rendered DOM from the server and client consistent, which is less likely to cause surprises for developers.

Reply:
Yeah that's fine, I don't see anyone wanting to override the width on a fullscreen dialog.